### PR TITLE
Add to Artifacts, and add optional arguments to entry points for flut…

### PR DESCRIPTION
**Pull request by keertip**
…ter run and test to allow for wiring up the same with preview-dart-2 internally

- add path to Dart VM for the engine, used to launch the frontend_server in dart 2
- add optional arguments for  passing in the path of the output dill file for dart 2 compilation
- pass in the packages file explicitly when it is specified
- for flutter test with dart 2 do not generate and compile the test wrapper file if a kernel file is specified

### CODE REVIEW

**Comment by keertip** [comment]
@tvolkert

**Comment by tvolkert** [comment]
Looks like Travis isn't happy

**Comment by keertip** [comment]
@tvolkert, this test failure

 RUNNING: cd dev/devicelab; ../../bin/cache/dart-sdk/bin/dart --checked --preview-dart-2 test/all.dart
Unhandled exception:
Unhandled exception:
type 'List' is not a subtype of type 'Iterable<String>' of 'variables' where
  List is from dart:core
  Iterable is from dart:core
  String is from dart:core
#0      new All (package:boolean_selector/src/all.dart)
#1      new Metadata._unresolved (package:test/src/backend/metadata.dart:153:26)
#2      new Metadata (package:test/src/backend/metadata.dart:167:59)
#3      new Declarer (package:test/src/backend/declarer.dart:109:29)
#4      _declarer (package:test/test.dart:58:25)
#5      group (package:test/test.dart:207:3)
#6      main (file:///Users/travis/build/flutter/flutter%20sdk/dev/devicelab/test/adb_test.dart:13:3)
#7      main (file:///Users/travis/build/flutter/flutter%20sdk/dev/devicelab/test/all.dart:10:12)
#8      _startIsolate.<anonymous closure> (dart:isolate/isolate_patch.dart:279:19)
#9      _RawReceivePortImpl._handleMessage (dart:isolate/isolate_patch.dart:165:12)
error: error evaluating constant constructor'package:test/src/backend/platform_selector.dart': error: line 25 pos 63: Not a constant expression.
  static const all = const PlatformSelector._(BooleanSelector.all);
                                                              ^
'package:test/src/backend/metadata.dart': error: line 198 pos 52: Not a constant expression.
      : testOn = testOn == null ? PlatformSelector.all : testOn,
                                                   ^
I can reproduce this in a clean checkout with no changes.

**Comment by tvolkert** [comment]
The Dart2 test failures are allowed on Travis and are just advisory.  The failures that are blocking are the following:

```
Analyzing 1410 files...
[hint] Unused import: 'base/file_system.dart'. (packages/flutter_tools/lib/src/compile.dart, line 12, col 8)
[hint] Unused import: 'base/common.dart'. (packages/flutter_tools/lib/src/compile.dart, line 11, col 8)
[lint] 18 public members lack documentation (ran in 111.2s)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
ERROR: Last command exited with 1 (expected: zero).
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 1]
```diff
@@ -252,6 +257,8 @@ class LocalEngineArtifacts extends Artifacts {
         return fs.path.join(_hostEngineOutPath, 'gen', _artifactToFileName(artifact));
       case Artifact.engineDartSdkPath:
         return fs.path.join(_hostEngineOutPath, 'dart-sdk');
+      case Artifact.engineDartBinary:
+        return fs.path.join(_hostEngineOutPath, 'dart-sdk','bin', _artifactToFileName(artifact));
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/artifacts.dart]: Nit: space after comma

**Review by tvolkert — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
Looks like this caused build breakage.

**Pull request merged**